### PR TITLE
fix(ui): fixing TopAppBar glitch during navigation

### PR DESCRIPTION
**Pull request by alaa-alsalehi**
After analyze #721 issue it looks like the reason of issue is the animation transition so I add a new class for animation in core module and use it in features modules

PART OF GTC OPEN SOURCE INTIATIVE

### CODE REVIEW

**Comment by alaa-alsalehi** [comment]
I reformat the file with issue @dturner

File: app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt [hunk 1]
```diff
@@ -196,14 +196,15 @@ fun NiaApp(
...
                         }
-
-                        NiaNavHost(appState = appState, onShowSnackbar = { message, action ->
-                            snackbarHostState.showSnackbar(
-                                message = message,
-                                actionLabel = action,
-                                duration = Short,
-                            ) == ActionPerformed
-                        })
+                        NiaAnimation {
```
Comment (tunjid) [inline app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt:199]: It's not quite clear how adding an animation on the snackbar solves an issue on the topAppBar. Can you please adda gif or a video showing your fix?

Comment (alaa-alsalehi) [reply app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt:199]: ACTUALLY THE GLITCH HERE IS AN INSTANT CHANGE OF PICTURE SO THE HUMAN EYE CATCH THIS DISCONTINUITY ADDING ANIMATION WILL MAKE THE PICTURE MOVING SMOOTHLY FOR THE HUMAN EYE AND MAKE IT CONTINUES
HERE IS A VIDEO TO SHOW THE NEW SITUATION
[device-2023-08-05-102456.webm](https://github.com/android/nowinandroid/assets/1716129/f501983a-f109-4bbb-86d8-bf3f17019a7d)

Comment (qamarelsafadi) [reply app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt:199]: @tunjid any thing we shall take into consideration here?

**Comment by keyboardsurfer** [comment]
/
Thank you for submitting this pull request. We have decided not to accept it at this time.
Feel free to comment or open an issue for further discussion if necessary.

**Pull request closed without merging**